File: Programmers/12921.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;
int cnt = 0;
int findPrime(int n) {
    for(int i = 2; i <= n; i++) {
        bool flag = true;
        for(int j = 2; j*j <= i; j++) {
            if (i%j == 0) {
                flag = false;
                break;
            }
        }
        if (flag) cnt++;
    }
    return cnt;
}

int solution(int n) {
    int answer = 0;
    findPrime(n);
    answer = cnt;
    return answer;
}

int main() {

    cout << solution(10) << endl;

    return 0;
}

// 어려운 프라임넘버..

File: Python/programmers0-1/12921.py
def solution(n):
    answer = 0
    s = set(range(2, n+1))
    for i in range(2, n+1):
        if i in s:
            s -= set(range(2*i, n+1, i))
    answer = len(s)
    return answer

'''
# 효율성 실패

def solution(n):
    answer = 0
    for i in range(2, n + 1):
        flag = True
        for j in range(2, i):
            if i % j == 0:
                flag = False
                break
            if j * j > i:
                break
        if flag:
            answer += 1
    return answer

'''